client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -265,6 +265,22 @@
             });
         };
 
+        this.saveWorkConfirmation = function saveWorkAuthoring(orig, diff, isDirty, message) {
+            var promise = $q.when();
+            if (isDirty) {
+                if (this.isEditable(diff)) {
+                    promise = confirm.confirmSaveWork(message)
+                        .then(angular.bind(this, function save() {
+                            return this.saveWork(orig, diff);
+                        }), function(err) { // cancel
+                            return $q.when();
```

Comment:
it would always call `onsuccess` in:
```
saveWorkConfirmation().then(onsuccess);
```
even if there is an error. so there should be `$q.reject()` or you can just remove that error handler.